framework/src/main/java/org/checkerframework/framework/util/element/ElementAnnotationUtil.java

```diff
@@ -435,6 +454,19 @@ private static AnnotatedTypeMirror getLocationTypeADT(
             enclosing = enclosing.getEnclosingType();
         }
 
+        int numOfEnclosingTypes = outerToInner.size() - 1;
+
+        // If the AnnotatedDeclaredType is a component of an array type, then apply anno to all
+        // inner types of the given declared type if the declared type has enclosing classes
```

Comment:
"to all possible inner types."

Add link to https://bugs.openjdk.java.net/browse/JDK-8208470 and state that this fix can be removed once JDK bug is fixed